cmd/juju/action/run.go

```diff
@@ -133,53 +133,54 @@ func (c *runCommand) SetFlags(f *gnuflag.FlagSet) {
 func (c *runCommand) Info() *cmd.Info {
 	return &cmd.Info{
 		Name:    "run-action",
-		Args:    "<unit> <action name> [key.key.key...=value]",
+		Args:    "<unit> [<unit> ...] <action name> [key.key.key...=value]",
 		Purpose: "Queue an action for execution.",
 		Doc:     runDoc,
 	}
 }
 
-// Init gets the unit tag, and checks for other correct args.
+// Init gets the unit tag(s), action name and action arguments.
 func (c *runCommand) Init(args []string) error {
-	switch len(args) {
-	case 0:
+	var unitNames []string
+	for idx, arg := range args {
+		if names.IsValidUnit(arg) {
```

Comment:
my concern on introspection of the args, is that I would guess people will eventually want the "run this on all units of an application", in which case 'IsValidUnit' no longer works.

I suppose if we wanted to make run-action act more like 'juju run' then we would have to supply flags always (juju run --application=/--machine=/--unit=). Would it be reasonable to support that in lieu of supporting more than one unit as a positional argument?

So we could allow:
  juju run-action --units postgresql/0,postgreqsl/1 action blah
which leaves us open to doing:
 juju run-action --application postgresql action blah
in the future

Thoughts?

Ultimately, I feel like 'juju run' and juju run-action *not* being nearly identical is a misfeature. Or arguably it should just be "juju run ... ACTION", or if we had to: "juju run --action ACTION"

Given that they both work via the same mechanism under the hood (AIUI), would it be possible to move more functionality into just the "juju run" command? Is there a reason to have 2 commands?